### PR TITLE
Fmod

### PR DESCRIPTION
**Pull request by asgibbons**

### CODE REVIEW

File: src/hotspot/cpu/x86/stubGenerator_x86_64_fmod.cpp [hunk 1]
```diff
@@ -0,0 +1,546 @@
+/*
+ * Copyright (c) 2016, 2021, 2023, Intel Corporation. All rights reserved.
```
Comment (sviswa7) [inline src/hotspot/cpu/x86/stubGenerator_x86_64_fmod.cpp]: This is a new file. Copyright (c)  2023, Intel Corporation. All rights reserved.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.cpp [hunk 2]
```diff
@@ -3937,6 +3937,10 @@ void StubGenerator::generate_initial_stubs() {
   }
 
   generate_libm_stubs();
+
+  if (VM_Version::supports_avx512vlbwdq() || VM_Version::supports_fma()) {
```
Comment (sviswa7) [inline src/hotspot/cpu/x86/stubGenerator_x86_64.cpp]: Good to check for UseAVX >= 2 as well:
if (VM_Version::supports_avx512vlbwdq() || (UseAVX >= 2) && (VM_Version::supports_fma())) {

File: src/hotspot/cpu/x86/sharedRuntime_x86.cpp [hunk 3]
```diff
@@ -97,11 +98,19 @@ jne    1b        \n\
...
+    jdouble dy = (jdouble) y;
+
+    retval = (jfloat) (*addr)(dx, dy);
+  }
   return retval;
 JRT_END
 
 JRT_LEAF(jdouble, SharedRuntime::drem(jdouble x, jdouble y))
   jdouble retval;
+  if (UseAVX < 1) {
```
Comment (sviswa7) [inline src/hotspot/cpu/x86/sharedRuntime_x86.cpp]: Does the new stub work for AVX=1 also or does it need AVX = 2 at the minimum?
Also good to check for UseFMA as well here.

File: src/hotspot/cpu/x86/sharedRuntime_x86.cpp [hunk 4]
```diff
@@ -112,6 +121,11 @@ jne    1b        \n\
     :"=t"(retval)
     :"0"(x), "u"(y)
     :"cc", "ax");
+  } else {
+    jdouble (*addr)(jdouble, jdouble) = (double (*)(double, double))StubRoutines::fmod();
```
Comment (sviswa7) [inline src/hotspot/cpu/x86/sharedRuntime_x86.cpp:129]: An assert here will be good to add that StubRoutines::fmod is not null. Or change the if condition above to have that check.

File: src/hotspot/cpu/x86/stubGenerator_x86_64_fmod.cpp (hunk 1 above)
Comment (asgibbons) [reply src/hotspot/cpu/x86/stubGenerator_x86_64_fmod.cpp]: Done.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.cpp (hunk 2 above)
Comment (asgibbons) [reply src/hotspot/cpu/x86/stubGenerator_x86_64.cpp]: Works for UseAVX == 1, so added condition UseAVX >= 1.

File: src/hotspot/cpu/x86/sharedRuntime_x86.cpp (hunk 3 above)
Comment (asgibbons) [reply src/hotspot/cpu/x86/sharedRuntime_x86.cpp]: Added !UseFMA for condition for which x87 code is used.

File: src/hotspot/cpu/x86/sharedRuntime_x86.cpp (hunk 4 above)
Comment (asgibbons) [reply src/hotspot/cpu/x86/sharedRuntime_x86.cpp:129]: Added an assert.

**Pull request closed without merging**